pkg/portal/cluster/statistics.go

```diff
@@ -0,0 +1,115 @@
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"time"
+
+	prometheusAPI "github.com/prometheus/client_golang/api"
+	v1 "github.com/prometheus/client_golang/api/prometheus/v1"
+	"github.com/prometheus/common/model"
+)
+
+type MetricValue struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     float64   `json:"value"`
+}
+
+type Metrics struct {
+	Name  string        `json:"metricname"`
+	Value []MetricValue `json:"metricvalue"`
+}
+
+func (c *client) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	return c.fetcher.Statistics(ctx, httpClient, promQuery, duration, endTime)
+}
+
+func (f *realFetcher) Statistics(ctx context.Context, httpClient *http.Client, promQuery string, duration string, endTime time.Time) ([]Metrics, error) {
+	promConfig := prometheusAPI.Config{
+		Address: "http://prometheus-k8s-0:9090",
+		Client:  httpClient,
+	}
+
+	client, err := prometheusAPI.NewClient(promConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	v1api := v1.NewAPI(client)
+
+	startTime, err := getStartTimeFromDuration(duration, endTime)
+	if err != nil {
+		return nil, err
+	}
+	value, warning, err := v1api.QueryRange(ctx, promQuery, v1.Range{
+		Start: startTime,
+		End:   endTime,
+		Step:  time.Minute * 2,
+	})
+	if len(warning) > 0 {
+		f.log.Warn(warning)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	valueMatrix := value.(model.Matrix)
+	return convertToStringMap(valueMatrix), nil
+}
+
+func convertToStringMap(v model.Matrix) []Metrics {
+	metrics := make([]Metrics, 0)
+	for _, i := range v {
+		metric := Metrics{}
+		metric.Name = i.Metric.String()
+		metricValues := make([]MetricValue, 0)
+		for _, j := range i.Values {
+			metricValues = append(metricValues, MetricValue{
+				Timestamp: j.Timestamp.Time().Local().UTC(),
+				Value:     float64(j.Value),
+			})
+		}
+		metric.Value = metricValues
+		metrics = append(metrics, metric)
+	}
+
+	return metrics
+}
+
+func getStartTimeFromDuration(duration string, endTime time.Time) (time.Time, error) {
+	switch duration {
+	case "1m":
+		return endTime.Add(-1 * time.Minute), nil
+	case "5m":
+		return endTime.Add(-5 * time.Minute), nil
+	case "10m":
+		return endTime.Add(-10 * time.Minute), nil
+	case "30m":
+		return endTime.Add(-30 * time.Minute), nil
+	case "1h":
+		return endTime.Add(-1 * time.Hour), nil
+	case "2h":
+		return endTime.Add(-2 * time.Hour), nil
+	case "6h":
+		return endTime.Add(-6 * time.Hour), nil
+	case "12h":
+		return endTime.Add(-12 * time.Hour), nil
+	case "1d":
+		return endTime.Add(-24 * time.Hour), nil
+	case "2d":
+		return endTime.Add(-48 * time.Hour), nil
+	case "1w":
+		return endTime.Add(-7 * 24 * time.Hour), nil
+	case "2w":
+		return endTime.Add(-14 * 24 * time.Hour), nil
+	case "4w":
+		return endTime.Add(-28 * 24 * time.Hour), nil
+	case "8w":
+		return endTime.Add(-56 * 24 * time.Hour), nil
+	}
+	return time.Time{}, errors.New("invalid duration")
+}
```

Comment:
I think `time.ParseDuration` would help you quite a bit here

Reply:
`time.ParseDuration` does not take `1d`, `1w` into account .. I can make the frontend to send the string in hours like for 1d send request for 24h but any place either frontend or backend .. I will have to do this switch case thing with ParseDuration, a little less but  will have to do that.

Reply:
I do not understand what you're trying to say

Reply:
we should probably use a standard duration format that is parsable by golang instead of having things like that

Reply:
okay done. I am converting date into hours at the frontend itself now and then sending to the backend so that ParseDuration() fucntion can identify the time.